File: mocks/oppgave-mock/src/main/java/no/nav/pensjon/vtp/mocks/oppgave/infrastruktur/validering/IsoDateValidator.java
package no.nav.pensjon.vtp.mocks.oppgave.infrastruktur.validering;

import static org.springframework.util.StringUtils.isEmpty;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

@SuppressWarnings("WeakerAccess")
public class IsoDateValidator implements ConstraintValidator<IsoDate, String> {

    @Override
    public void initialize(IsoDate isoDate) {
        //JSR303
    }

    @Override
    public boolean isValid(String date, ConstraintValidatorContext constraintValidatorContext) {
        if (!isEmpty(date)) {
            try {
                LocalDate.parse(date, DateTimeFormatter.ISO_LOCAL_DATE);
            } catch (Exception e) {
                return false;
            }
        }

        return true;
    }

}

File: mocks/oppgave-mock/src/main/kotlin/no/nav/pensjon/vtp/mocks/oppgave/rest/infrastruktur/validering/IsoDateValidator.kt
package no.nav.pensjon.vtp.mocks.oppgave.rest.infrastruktur.validering

import org.springframework.util.ObjectUtils.isEmpty
import java.time.LocalDate.parse
import java.time.format.DateTimeFormatter.ISO_LOCAL_DATE
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class IsoDateValidator : ConstraintValidator<IsoDate?, String?> {
    override fun initialize(isoDate: IsoDate?) {
        // JSR303
    }

    override fun isValid(date: String?, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        return when {
            isEmpty(date) -> true
            else ->
                try {
                    parse(date, ISO_LOCAL_DATE)
                    true
                } catch (e: Exception) {
                    false
                }
        }
    }
}
